### ISSUE TITLE
Script stopped working today

### ISSUE THREAD

**Issue by Kalytis** [pixeltris/TwitchAdSolutions#236]
Script used to work perfectly fine, but since this morning, I have pre/mid-roll ads again. Not sure what changed.

**Comment by iNfz** [comment]
yeah. ads are coming through

**Comment by Kalytis** [comment]
Well, today it seemed to be working fine... It's mind-boggling.

**Comment by TuffestLeader** [comment]
not working again :(

**Comment by xandertreat** [comment]
Yep, stopped today

**Comment by dragothica** [comment]
For me Twitch stopped loading the video stream completely today. I also had been using a Firefox Extension "Twitch LOL Pro" (adblocker). Disabling this extension helped and now Twitch is working again. I also have enabled Twitch Ad Solutions via Userscriopt (Ublock Origin).

**Comment by dragothica** [comment]
Yeah I was curious just mentioning it. Probably should just unistall it as it hasn't been updated in ages. Twitch was working though previously so I had left it on.

**Comment by hajo808** [comment]
Yes, the same for me. Since today, nothing has worked at all. I only use Ublock org. error #2000

**Comment by pixeltris** [comment]
It's always best to mention what script you are using when saying something doesn't work as the 2 scripts in this repo do behave slightly differently. As for other solutions you want to go to the issues of the given project. In the case of `TTV LOL Pro` it is https://github.com/younesaassila/ttv-lol-pro/issues

That being said. I currently can't reproduce. There is also an open issue #225

**Comment by salvadorzombie** [comment]
Both of the scripts (vaft and video-swap-new) stopped working for me last night. I've been using video-swap-new and at some time between 11pm and 1am (best guess) I first got an ad on Twitch. Usually when one gives me trouble (maybe every 4-5 months) I swap to the other and it works fine. Neither was working until a few hours ago, when video-swap-new suddenly started blocking again. *Just now* about five minutes ago, I got another ad while using video-swap-new, and swapping to vaft isn't working either).

**Comment by Tammonll** [comment]
> Both of the scripts (vaft and video-swap-new) stopped working for me last night. I've been using video-swap-new and at some time between 11pm and 1am (best guess) I first got an ad on Twitch. Usually when one gives me trouble (maybe every 4-5 months) I swap to the other and it works fine. Neither was working until a few hours ago, when video-swap-new suddenly started blocking again. _Just now_ about five minutes ago, I got another ad while using video-swap-new, and swapping to vaft isn't working either).

Same issue here. For me it started 5 hours ago aswell and getting constant midroll ads

**Comment by salvadorzombie** [comment]
The weirdest part is that, like I mentioned, sometime a little while ago the "Blocking ads" popped up, I got the proxy 480p feed, and it was working fine. Worked through multiple channels, too. Just now about a half hour ago it stopped working again.

**Comment by Tammonll** [comment]
i hope they come up with a fix soon.. every 15 min a add. I rather watch the purple screen again

**Comment by elderddragon** [comment]
Just to add info, I've used the scripts (both vaft and swap) for months in uBlock. This past week it stopped working on uBlock, so I started using the userscript with violentmonkey as the manager, and it went back to working. But as of the past 36 hours to 48 hours  (April 8th or 9th) it has stopped working as well. Any ads go through immediately. 

I hope this info can help with anything, if at all.

**Comment by elderddragon** [comment]
I don't know if links are allowed, but I used both vaft and video-swap-new with violentmonkey (not at the same time, I would disable/uninstall one before using the other).

**Comment by salvadorzombie** [comment]
Also, I should specify ahead of time that I'm on Firefox.

**Comment by pixeltris** [comment]
> Just to add info, I've used the scripts (both vaft and swap) for months in uBlock. This past week it stopped working on uBlock, so I started using the userscript

It is mentioned in the readme now that uBlock Origin randomly stops applying the script. Userscript version is recommended. Ref #200

**Comment by salvadorzombie** [comment]
I've tried the userscript version today multiple times (I'm trying it right now again in fact) but it's not working, either.

**Comment by pixeltris** [comment]
@salvadorzombie did you try tampermonkey? I think I saw ads on one of the script managers and didn't bother to look into fixing it. But tampermonkey definitely works with the scripts.

**Comment by salvadorzombie** [comment]
Tampermonkey is the Chrome version, Greasemonkey is the version for Firefox. I'm on Firefox.

**Comment by salvadorzombie** [comment]
> Yeah, I forgot to mention. I'm on Firefox, using userscripts now with violentmonkey as the manager, instead of uBlock. It's not working.
> 
> I'll try tampermonkey then.

If you're on Firefox you can't use Tampermonkey, because Tampermonkey is specifically for Chrome. Greasemonkey is the version on Firefox.

**Comment by salvadorzombie** [comment]
Actually, I just read up again and apparently Greasemonkey on FF was the original and Tamper was the Chrome version (and Violent was the Opera version), but they're all on Firefox now. Getting ready to try the userscript on Tamper, wish me luck!

**Comment by salvadorzombie** [comment]
At first I got an ad with Tampermonkey immediately, but a refresh got rid of it. I'm checking other channels and neither an ad nor the "Blocking Ads" message have popped up on any, so I'm not sure if it's working or if it just counted the time I had it on as watching the ad. I'll keep trying, once I see a "Blocking Ads" message I'll be convinced it's working.

**Comment by pixeltris** [comment]
Weird. You can check if the script is being injected by checking the console log for `hookWorkerFetch`. If it doesn't get logged it isn't being injected. Or typing in the console `window.reloadTwitchPlayer`, if it shows up as `undefined` then the script isn't being injected. Unfortunately I can't help in terms of the script not being injected as that is out of my control. If it is being injected you could give logs of the m3u8 files and I could see why ads are slipping through.

**Comment by zGato** [comment]
This could be helpful to add, but I doubt.
https://twitch.tv/twitchmedia99 will trigger a preroll everytime worldwide, neither VAFT or video-swap-new is able to bypass it.

I guess that stream just doesn't have the mini player/360p player available.

![image-2.png](https://github.com/pixeltris/TwitchAdSolutions/assets/88756568/cf4d11f2-8e42-4622-ac68-affdac925bbc)

**Comment by salvadorzombie** [comment]
Okay...this is really weird. It just started working again.

It wasn't working on Tamper, so I removed that, and decided to try the uBO way again. I had been using the "ublock (permalink)" URL for that ever since I started using this method (at least for the last year or two). This time, I used the URL *in the example* (which I think is the "ublock" one that's listed as one of the three options, the middle one). Worked on the first try.

I genuinely have no idea if this fixed it or if something else happened, but I'll check it again with a few more channels just to be sure. (In case this info is needed at a later date - I'm on Firefox, using uBO method, using the "ublock" URL, and I'm on Windows 7 which I know is an extreme edge case but I'm just unable to build a computer right now.)

**Comment by salvadorzombie** [comment]
And........nevermind.

The channel I was on was stuck on *permanent* "Blocking Ads" message. The second I went to another channel, immediately got an ad.

**Comment by salvadorzombie** [comment]
Also, I'm not experienced in any of this, so I have no idea how to do a lot of this. I know how to access the Browser Console on FF and I can also rightclick -> Inspect, but I have no idea if either of those is what you mean. Couldn't find window.reloadTwitchPlayer on either one, though.

**Comment by salvadorzombie** [comment]
I just tried vaft using the middle URL for the uBO method, and it seemed like it was going to go permanent "Blocking Ads" like the other, but it cleared up after a few minutes and then re-sizing the video (from using Inspect on the page). Checked a second channel, "Blocking Ads" pops up again (for some reason vaft seems to either not go to 480p or waver between 480p and what I have selected), and then cleared after about 30 seconds, like normal. It seems like vaft is working for me, now, using that middle URL.

**Comment by elderddragon** [comment]
All tests on FireFox, latest version. 

I tried both vaft and video-swap, using greasemonkey, tampermonkey, violentmonkey. Ads are playing without problem, it is like nothing tries to stop them.

I just read a comment on testing if the script is being injected, so I'll try that. I'll edit the comment to avoid more clutter here.

**Comment by salvadorzombie** [comment]
> All tests on FireFox, latest version.
> 
> I tried both vaft and video-swap, using greasemonkey, tampermonkey, violentmonkey. Ads are playing without problem, it is like nothing tries to stop them.
> 
> I just read a comment on testing if the script is being injected, so I'll try that. I'll edit the comment to avoid more clutter here.

I've found something that seems to be working for me. I just tried a third channel that got a "Blocking Ads" message, and while it froze after about 30 seconds, refreshing using the FFZ Reset Player button cleared it up and the message is gone. So I think this is mostly working (sometimes the resolution wavers and it freezes sometimes during the ad break but the ads aren't coming through).

I'm using the "ublock"vaft URL (the one in the middle of the three) using the uBO script method. On current Firefox, using the most current version of uBO (1.57.2, I literally updated it about 2 hours ago, check to see if there's an update). No idea if it'll work for you but it's currently working for me.

**Comment by salvadorzombie** [comment]
> This could be helpful to add, but I doubt. https://twitch.tv/twitchmedia99 will trigger a preroll everytime worldwide, neither VAFT or video-swap-new is able to bypass it.
> 
> I guess that stream just doesn't have the mini player/360p player available.
> 
> ![image-2.png](https://private-user-images.githubusercontent.com/88756568/321309616-cf4d11f2-8e42-4622-ac68-affdac925bbc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3NzM2NzIsIm5iZiI6MTcxMjc3MzM3MiwicGF0aCI6Ii84ODc1NjU2OC8zMjEzMDk2MTYtY2Y0ZDExZjItOGU0Mi00NjIyLWFjNjgtYWZmZGFjOTI1YmJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDEwVDE4MjI1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMDIxNDVkM2ZkMTBlMGU3YzMxMWQ3MTM2NWQ4YzQ3MjcwMGQxOThhOGFjZTViYzE2MDQxNGMzMjYxZGJjZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.X1qt2mjbuBhkURYdYMRseXg2m6LfMLWNO2E3Xg6rkiI)

I looked on the page that has your list and there's something called "screwtwitchads" that says it's not currently released. Will that be available in the future? From the description it sounds like it's working perfectly, which would be great given that nothing I'm trying now seems to work.

**Comment by salvadorzombie** [comment]
Also just something I was thinking about - everything was fine until last night, then it was fine for a while and then broke again.

uBO apparently had an update at some point between last night and today, and then there was another update after the first one, and both *roughly* coincide with all of these failing to work. Could it be that there's been something in uBO to now cause the scripts to fail?

**Comment by JimmyBoyHalpert** [comment]
> For me Twitch stopped loading the video stream completely today. I also had been using a Firefox Extension "TTV LOL Pro" (adblocker). Disabling this extension helped and now Twitch is working again. I also have enabled Twitch Ad Solutions via Userscriopt (Ublock Origin).

Came to say the same thing. No video could manage to load for me. I forgot I had TTV LOL Pro installed while also having the userscript through uBlock. I disabled TTV LOL Pro and everything is working just fine. Instead of ads, I just have the low resolution video in place of when an ad would appear. Unfortunately seems TTV LOL Pro  will be broken for people.

**Comment by salvadorzombie** [comment]
> > For me Twitch stopped loading the video stream completely today. I also had been using a Firefox Extension "TTV LOL Pro" (adblocker). Disabling this extension helped and now Twitch is working again. I also have enabled Twitch Ad Solutions via Userscriopt (Ublock Origin).
> 
> Came to say the same thing. No video could manage to load for me. I forgot I had TTV LOL Pro installed while also having the userscript through uBlock. I disabled TTV LOL Pro and everything is working just fine. Instead of ads, I just have the low resolution video in place of when an ad would appear. Unfortunately seems TTV LOL Pro will be broken for people.

Yeah, this doesn't help me. I made sure to remove everything but the single script/add-on I was trying. Nothing's working for me at all for some reason.

**Comment by zGato** [comment]
> @salvadorzombie I looked on the page that has your list and there's something called "screwtwitchads" that says it's not currently released. Will that be available in the future? From the description it sounds like it's working perfectly, which would be great given that nothing I'm trying now seems to work.

It used to be quite simple to just install and forget but Twitch partially broke it this 2024. It does still work just fine, but it breaks every week or so, and I don't feel like creating a script to automate it.

I just use TTV LOL PRO personally as I don't want to deal with scripts breaking or some other random shit when I barely watch Twitch at all, I just want a clear stream when I want it, that's it.

"nothing I'm trying seems to work" is not exactly true, TTV LOL PRO still works just fine.

**Comment by salvadorzombie** [comment]
> > @salvadorzombie I looked on the page that has your list and there's something called "screwtwitchads" that says it's not currently released. Will that be available in the future? From the description it sounds like it's working perfectly, which would be great given that nothing I'm trying now seems to work.
> 
> It used to be quite simple to just install and forget but Twitch partially broke it this 2024. It does still work just fine, but it breaks every week or so, and I don't feel like creating a script to automate it.
> 
> I just use TTV LOL PRO personally as I don't want to deal with scripts breaking or some other random shit when I barely watch Twitch at all, I just want a clear stream when I want it, that's it.
> 
> "nothing I'm trying seems to work" is not exactly true, TTV LOL PRO still works just fine.

I know it works for you, but I tried TTV LOL PRO (both v1 and v2) and it's not working at all for me. At this point I just want something to even show a hint of working, everything stopped working completely like 3 hours ago.

**Comment by zGato** [comment]
> @salvadorzombie I know it works for you, but I tried TTV LOL PRO (both v1 and v2) and it's not working at all for me. At this point I just want something to even show a hint of working, everything stopped working completely like 3 hours ago.

As one of the devs of TTV LOL PRO I can certainly say it's a you issue, TTV LOL PRO is working just fine for everyone, literally. 
Read the wiki and understand how it works/recommended extensions to use alongside, ...

**Comment by salvadorzombie** [comment]
(I was editing my last post when I saw you post so I'll just add that here to this.)

I genuinely believe you when you say it's working. I'm sure it's working for like 99.9% of people. Hell, the video-swap-new and vaft worked perfectly for me 99.9% of the time for the last 2-3 years. I really did read everything there and followed the instructions, and for some reason it's not working for me. Not even a "Commercial Break" screen.

I'm really sorry, I know I have to seem like an insane person right now, but I can't think of any way to get any of these work. Every single one seems to just *not exist* when I install it. I make sure that each one is the only thing running, I cleared my cache last night, I'm following the instructions to the letter and I read the whole Must Read section for TTV LOL PRO just to make sure I didn't miss something. Nothing seems to work. I have no idea why and I don't know what to do.

I know this isn't anyone's fault and I really appreciate the work everyone does to make these and maintain them. It's just making me feel crazy not being able to fix this even after I follow all of the instructions. I'm really sorry, I know I have to seem like an insane person right now, but I can't think of any way to get any of these work. Every single one seems to just *not exist* when I install it. I make sure that each one is the only thing running, I cleared my cache last night, I'm following the instructions to the letter and I read the whole Must Read section for TTV LOL PRO just to make sure I didn't miss something. Nothing seems to work. I have no idea why and I don't know what to do.

**Comment by zGato** [comment]
@salvadorzombie just DM me on Discord and I can personally help you out
I genually think you have good intentions.
`zgato_` (or zGato | Marc#7399)

**Comment by salvadorzombie** [comment]
Thanks, I just sent you a friend request. I really appreciate it.

**Comment by fREAST** [comment]
Using the vaft userscript, its seems to be working in Firefox, during ads its replaced with the lower quality stream, but in Chrome it doesn't seem to work, the ads play.
I can see in the console it says `Twitch adblocker is enabled` so the script should be loaded.

**Comment by filipemanuelofs** [comment]
I was using `video-swap-new-ublock-origin.js` without any problem until today. Noticed some ads and replaced it with the old `vaft-ublock-origin.js` and now the ads are gone. I go back and forth between `vaft` and `video-swap`, not the end of the world.

**Comment by whaaaley** [comment]
> > @salvadorzombie I know it works for you, but I tried TTV LOL PRO (both v1 and v2) and it's not working at all for me. At this point I just want something to even show a hint of working, everything stopped working completely like 3 hours ago.
> 
> As one of the devs of TTV LOL PRO I can certainly say it's a you issue, TTV LOL PRO is working just fine for everyone, literally. Read the wiki and understand how it works/recommended extensions to use alongside, ... "Commercial break in progress" with the public proxies (Default provided) is expected on peak hours, that is not "not working"

Hey, Just wanted to mention that your extension stopped working for me this week as well. Twitch releases often come out slowly over time. More people will start having issues soon.

**Comment by unknownids** [comment]
firefox
tampermonkey
TwitchAdSolutions (vaft) 6.0

I had an ad show on a stream just now. Closed off the tab. Waited a minute, opened the stream again and got the standard blocking ad message. So there is no consistency and I've noticed that over the past few days when this stopped working (sort of). It will sometimes show an ad, other times will block it which makes this even worse to trouble shoot.

**Comment by Omarock92** [comment]
Both vaft and videoswap stopped working for me today.

Chrome 123.0.6312.123
Ublock origin

**Comment by TsukiZero** [comment]
I tried every single solution and none of them works. Not even the ad hider hides them until it ends.

**Comment by Tammonll** [comment]
yeah it's fubar atm :( Just got a 3 min movie trailer ad...

**Comment by PocoShin** [comment]
Same here, using both scripts, plus the box above the chat.
![Ads](https://github.com/pixeltris/TwitchAdSolutions/assets/71533923/fe66fbc3-3942-41e5-bed0-f96b361816b6)

**Comment by hajo808** [comment]
Strange, since a few hours I get a picture of the stream again instead of a black screen.
I had set uBlock to "unset" again two days ago. Then I opened Chrome with a black screen to write in the stream and opened a private Chrome window to watch the stream. I noticed that there was only one ad at the beginning of the stream. Then not at all. But since today I can also watch Twitch again with the normal Chrome browser window. When I switched streams, there were no ads until now. I have no idea. By the way, I still use Pi-hole and unbound. I will continue to monitor it. I also use the extensions 7TV, BetterTTV, Privacy Badger, TTV LOL and uBlock Origin.

**Comment by guestds** [comment]
im not sure if its useful or not, but using vaft and video-swap-new at the same time usually causes the stream to get locked on a black screen and never play, but its not doing so during a stream, it still "successfully" breaks in the clips page or on a stream preview on the twitch home page

**Comment by therentabrain** [comment]
I also started getting ads a few days ago.  

Using the vaft script in ublock.  I see "Twitch adblocker is enabled" in the console.

> window.reloadTwitchPlayer

This shows as 'undefined'.

Happy to provide m3u8 files if you can tell me where to find them!

**Comment by therentabrain** [comment]
Is this a clue?

`[Report Only] Refused to create a worker from 'blob:https://www.twitch.tv/b5210a1a-de66-47f8-8daf-0d3a468fb6ca' because it violates the following Content Security Policy directive: "worker-src 'none'".`

**Comment by pixeltris** [comment]
I can finally reproduce this. I'll take a proper look soon.

**Comment by abhishah901** [comment]
My temporary solution to this is Alternate Twitch Player. It blocks ads with a black screen, at least you don't see the pesky ads~

**Comment by therentabrain** [comment]
I was mucking about with it in Tampermonkey, put in a little log message to check, and indeed found that it never triggered.
```
    function hookWorkerFetch() {
        console.log('Twitch adblocker is enabled');
        var realFetch = fetch;
        fetch = async function(url, options) {
            console.log('adblocker got here');
```

**Comment by pixeltris** [comment]
Should be fixed in https://github.com/pixeltris/TwitchAdSolutions/commit/044d1fb3bb88a19ab718603069354d43eeba7015. If you're getting error `#1000`/`#2000` see #228 as that still hasn't been addressed.

**Comment by elderddragon** [comment]
I dont know the github "etiquette" but I just wanted to say thank you, pixeltris. this looks like a mostly thankless job. so yeah, thanks again.

**Comment by pixeltris-fan** [comment]
@pixeltris you fuckin rock dude. thank you so much for all the work you do. i second what elderddragon said. this might not be the right place to ask but is there some way for us to financially contribute to you?

**Comment by IPHANT0MI** [comment]
@pixeltris thank you so much! It's working now

**Comment by bronze-aged** [comment]
What an absolute legend...cannot believe you fixed this so quickly. Twitch would be absolutely unusable without your help!

**Comment by JonJaded** [comment]
You're the best, you're the best. Thanks.

**Comment by TsukiZero** [comment]
You SAVED us!

**Comment by Samuvasc** [comment]
I'm still having ads on twitch. What should I do?
The "say i am an advanced user and copypaste the link on the user resouces thingy" doesn't currently work. Please describe what a non programmer user must do.

**Comment by IPHANT0MI** [comment]
> I'm still having ads on twitch. What should I do? The "say i am an advanced user and copypaste the link on the user resouces thingy" doesn't currently work. Please describe what a non programmer user must do.

Did you make sure you are using the correct url? You want to make sure to be using the permalink ublock.

Once it’s put in place make sure you close out the browser completely (reboot PC) if you have too and try it again. Should work.

![IMG_2623](https://github.com/pixeltris/TwitchAdSolutions/assets/91971743/47d01c56-5cbb-404b-aa13-fc485f2dcd1b)

https://github.com/pixeltris/TwitchAdSolutions
